--- origsrc/xmlrpc-c-1.06.30/xmlrpc-c-config.in	2007-08-04 17:38:34.000000000 -0500
+++ src/xmlrpc-c-1.06.30/xmlrpc-c-config.in	2008-07-23 21:18:45.609375000 -0500
@@ -116,12 +116,12 @@
         the_wl_rpath="@WININET_WL_RPATH@ $the_wl_rpath"
       fi
       if test "${MUST_BUILD_CURL_CLIENT}" = "yes"; then
-        the_libs="@CURL_LDADD@ $the_libs"
+        the_libs="$the_libs @CURL_LDADD@"
         the_rpath="@CURL_RPATH@ $the_rpath"
         the_wl_rpath="@CURL_WL_RPATH@ $the_wl_rpath"
       fi
       if test "${MUST_BUILD_LIBWWW_CLIENT}" = "yes"; then
-        the_libs="@LIBWWW_LDADD@ $the_libs"
+        the_libs="$the_libs @LIBWWW_LDADD@"
         the_rpath="@LIBWWW_RPATH@ $the_rpath"
         the_wl_rpath="@LIBWWW_WL_RPATH@ $the_wl_rpath"
       fi
--- origsrc/xmlrpc-c-1.06.30/xmlrpc-c-config.test.in	2007-08-04 17:38:34.000000000 -0500
+++ src/xmlrpc-c-1.06.30/xmlrpc-c-config.test.in	2008-07-23 21:18:45.625000000 -0500
@@ -90,12 +90,12 @@
         the_wl_rpath="@WININET_WL_RPATH@ $the_wl_rpath"
         fi
       if test "${MUST_BUILD_CURL_CLIENT}" = "yes"; then
-        the_libs="@CURL_LDADD@ $the_libs"
+        the_libs="$the_libs @CURL_LDADD@"
         the_rpath="@CURL_RPATH@ $the_rpath"
         the_wl_rpath="@CURL_WL_RPATH@ $the_wl_rpath"
         fi
       if test "${MUST_BUILD_LIBWWW_CLIENT}" = "yes"; then
-        the_libs="@LIBWWW_LDADD@ $the_libs"
+        the_libs="$the_libs @LIBWWW_LDADD@"
         the_rpath="@LIBWWW_RPATH@ $the_rpath"
         the_wl_rpath="@LIBWWW_WL_RPATH@ $the_wl_rpath"
         fi
