--- origsrc/xmlrpc-c-1.06.30/src/cpp/value.cpp	2007-09-30 15:55:18.000000000 -0500
+++ src/xmlrpc-c-1.06.30/src/cpp/value.cpp	2008-07-23 21:18:45.593750000 -0500
@@ -32,6 +32,7 @@
 #include <string>
 #include <vector>
 #include <time.h>
+#include <sys/time.h>
 
 #include "xmlrpc-c/girerr.hpp"
 using girerr::error;
--- origsrc/xmlrpc-c-1.06.30/tools/xml-rpc-api2cpp/Makefile	2006-11-05 17:53:14.000000000 -0600
+++ src/xmlrpc-c-1.06.30/tools/xml-rpc-api2cpp/Makefile	2008-07-23 21:18:45.593750000 -0500
@@ -19,7 +19,7 @@
 
 LDFLAGS = $(LADD)
 
-LDLIBS = -Lblddir/src/cpp -lxmlrpc_cpp -lxmlrpc_server $(CLIENT_LDLIBS)
+LDLIBS = blddir/src/cpp/libxmlrpc_cpp.a -lxmlrpc_server $(CLIENT_LDLIBS)
 
 PROGS :=
 
